### ISSUE TITLE
Modifying ec2 security groups in ansible playbook doesn't work for running instances

### ISSUE THREAD

**Issue by kuza55** [ansible/ansible-modules-core#4083]
##### ISSUE TYPE
- Bug Report
##### COMPONENT NAME

ec2 module
##### ANSIBLE VERSION

$ ansible --version
ansible 2.1.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
##### OS / ENVIRONMENT

Ubuntu on Windows 10
##### SUMMARY

Changing the group_id value for a running instance has no effect on it's security groups
##### STEPS TO REPRODUCE

Run playbook that creates ec2 instance
Add extra group_id to playbook
Run playbook again
##### EXPECTED RESULTS

ec2 instance is in both security groups
##### ACTUAL RESULTS

security group of instance is not changed.

**Comment by nitzmahone** [comment]
@defionscode or somebody else correct me if I'm wrong, but IIRC AWS doesn't allow you to change the security group mapping of a running instance, only to make changes to the rules in the existing mapped group...

**Comment by nitzmahone** [comment]
NM, it appears they've lifted that restriction at some point: http://docs.aws.amazon.com/AmazonVPC/latest/UserGuide/VPC_SecurityGroups.html#SG_Changing_Group_Membership.

@kuza55 can you confirm that you're doing this to a VPC instance and not an EC2 Classic instance?

**Comment by kuza55** [comment]
Yep, doing this to a VPC instance. I ended up doing this from the AWS web UI and it worked.

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/jctanner/ansibullbot/blob/ISSUE_TRIAGE_CLASS/ISSUE_HELP.md)

**Comment by jctanner** [comment]
Possibly a duplicate of https://github.com/ansible/ansible-modules-core/issues/454

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
@ansible, ping. This issue is still waiting on your response.
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
This repository has been locked. All new issues and pull requests should be filed in https://github.com/ansible/ansible

Please read through the <a href="http://docs.ansible.com/ansible/dev_guide/repomerge.html#move-issues-and-prs-to-new-repo">repomerge</a> page in the dev guide. The guide contains links to tools which automatically move your issue or pull request to the ansible/ansible repo.

**Comment by ansibot** [comment]
This issue was migrated to https://github.com/ansible/ansible/issues/29762
